pod-operation/src/components/wheel_encoder.rs

```diff
@@ -1,70 +1,94 @@
-use rppal::gpio::{Gpio, InputPin, Level};
-use std::time::Instant;
+use tokio::time::Instant;
+use rppal::gpio::{Gpio, InputPin};
 
-const PIN_ENCODER_A: u8 = 1;
-const PIN_ENCODER_B: u8 = 2;
 
+// GPIO pins for the encoder
+const PIN_ENCODER_A: u8 = 23; //need to move
+const PIN_ENCODER_B: u8 = 24; //need to move
+
+// Constants for calculations
+const DELTA_D: f32 = 1.0 / 16.0;
+
+type EncoderState = u8;
+type EncoderDiff = i8;
```

Comment:
Thought: to be more specific with the values, we should be able to construct an enum with the `num_enum` traits. I'll look into this.